### ISSUE TITLE
Visual Scripting Addon - Background Crash

### ISSUE THREAD

**Issue by animatedai** [bradklingensmith/project-glue#2]
This seems to happen randomly. I believe this caused by Visual Scripting Addon doing something that doesn't always work when blender is launched in the background due to a bug in either the addon or Blender itself.

```
Error:
  stderr:
  Traceback (most recent call last):
    File "C:\Program Files\Blender Foundation\Blender 3.6\3.6\scripts\modules\addon_utils.py", line 369, in enable
      mod.register()
    File "C:\Users\Brad\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\blender_visual_scripting_addon\__init__.py", line 79, in register
      register_keymaps()
    File "C:\Users\Brad\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\blender_visual_scripting_addon\keymaps\keymap.py", line 22, in register_keymaps
      km = kc.keymaps.new(name="Node Editor", space_type="NODE_EDITOR")
  AttributeError: 'NoneType' object has no attribute 'keymaps'
  Traceback (most recent call last):
    File "C:\Users\Brad\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\AI-Render\handlers.py", line 21, in load_post_handler
      if context.scene.air_props.is_enabled:
  AttributeError: 'str' object has no attribute 'scene'
  Traceback (most recent call last):
    File "C:\Users\Brad\Documents\blender\project-manager\scripts\addons\Project GLUE\ui\execute_action_chain_operator.py", line 26, in execute
      action()
    File "C:\Users\Brad\Documents\blender\project-manager\scripts\addons\Project GLUE\refactoring\move_datablocks.py", line 37, in make_and_remap_target
      mybpy.save_as_mainfile(target_path)
    File "C:\Users\Brad\Documents\blender\project-manager\scripts\addons\Project GLUE\mybpy\__init__.py", line 69, in save_as_mainfile
      ops.wm.save_as_mainfile(filepath=file_path.path, check_existing=False, relative_remap=True)
    File "C:\Program Files\Blender Foundation\Blender 3.6\3.6\scripts\modules\bpy\ops.py", line 113, in __call__
      ret = _op_call(self.idname_py(), None, kw)
  RuntimeError: Error: Cannot open file C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Functions\Integer Math.blend \!\.blend@ for writing: No such file or directory
  
  
  During handling of the above exception, another exception occurred:
  
  Traceback (most recent call last):
    File "C:\Users\Brad\Documents\blender\project-manager\scripts\addons\Project GLUE\ui\execute_action_chain_operator.py", line 36, in execute
      error_message=error_handling.create_bug_report_essentials(e)
  AttributeError: module 'Project GLUE.error_handling' has no attribute 'create_bug_report_essentials'
  
  
  stdout:
  WARN (bgl): C:\Users\blender\git\blender-v360\blender.git\source\blender\python\generic\bgl.c:2662 BPyInit_bgl: 'bgl' imported without an OpenGL backend. Please update your add-ons to use the 'gpu' module. In Blender 4.0 'bgl' will be removed.
  Blender 3.6.1 (hash 8bda729ef4dc built 2023-07-17 23:35:37)
  Read prefs: "C:\Users\Brad\AppData\Roaming\Blender Foundation\Blender\3.6\config\userpref.blend"
  Read blend: "C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Pop-in Text.blend"
  Info: Read library:  'C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Create Text.blend', '//Create Text.blend', parent '<direct>'
  Info: Read library:  'C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Cube Tensor Unit Glass.blend', '//Cube Tensor Unit Glass.blend', parent '<direct>'
  Info: Read library:  'C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Functions - Shader.blend', '//Functions - Shader.blend', parent '<direct>'
  Info: Read library:  'C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Functions.blend', '//Functions.blend', parent '<direct>'
  Info: Read library:  'C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Get Calc Activation.blend', '//Get Calc Activation.blend', parent 'C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Create Text.blend'
  Info: Read library:  'C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Functions\Integer Math.blend', '//Functions\Integer Math.blend', parent '<direct>'
  Info: Read library:  'C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Theme.blend', '//Theme.blend', parent '<direct>'
  Info: Read library:  'C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Cube Tensor Unit Rubber.blend', '//Cube Tensor Unit Rubber.blend', parent 'C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Create Text.blend'
  Info: Read library:  'C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\PB HSL.blend', '//PB HSL.blend', parent 'C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Create Text.blend'
  Info: Read library:  'C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Cube Tensor Unit Metal.blend', '//Cube Tensor Unit Metal.blend', parent 'C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Theme.blend'
  Info: Read library:  'C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Shader - Get Tensor Color.blend', '//Shader - Get Tensor Color.blend', parent 'C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Cube Tensor Unit Rubber.blend'
  Error: Python: Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "C:\Users\Brad\Documents\blender\animated-ai\v2\.glue\expr.py", line 1, in <module>
      import bpy;bpy.ops.glue.execute_action_chain(make_actions_pickled="80049523020000000000008c0966756e63746f6f6c73948c077061727469616c9493948c2850726f6a65637420474c55452e7265666163746f72696e672e6d6f76655f64617461626c6f636b73948c256d616b655f6d6f76655f64617461626c6f636b5f747265655f616374696f6e5f636861696e949394859452942868058c2450726f6a65637420474c55452e64617461626c6f636b5f6e616d652e6162736f6c757465948c084162736f6c7574659493942981947d94288c0f626c656e645f66696c655f70617468948c1550726f6a65637420474c55452e706174682e616c6c94680993942981947d948c0470617468948c4b433a5c55736572735c427261645c446f63756d656e74735c626c656e6465725c616e696d617465642d61695c76325c7072696d6974697665735c506f702d696e20546578742e626c656e649473628c0c7261775f726e615f6e616d65948c1247656f6d65747279204e6f64652054726565948c0f6e61746976655f726e615f6e616d65948c1247656f6d65747279204e6f64652054726565948c0e64617461626c6f636b5f6e616d65948c13496e6372656d656e742028496e746567657229947562680f2981947d9468128c60433a5c55736572735c427261645c446f63756d656e74735c626c656e6465725c616e696d617465642d61695c76325c7072696d6974697665735c46756e6374696f6e735c496e7465676572204d6174682e626c656e64205c215c2e626c656e6494736286947d944e7494622e", guid="7be923de-315f-4f2a-a352-7d595bd11c84-2ae9c7c0-ddae-47cb-88a7-9f4cc548bd2f", progress_title="Moving datablock and updating users")
    File "C:\Program Files\Blender Foundation\Blender 3.6\3.6\scripts\modules\bpy\ops.py", line 113, in __call__
      ret = _op_call(self.idname_py(), None, kw)
  RuntimeError: Error: Python: Traceback (most recent call last):
    File "C:\Users\Brad\Documents\blender\project-manager\scripts\addons\Project GLUE\ui\execute_action_chain_operator.py", line 26, in execute
      action()
    File "C:\Users\Brad\Documents\blender\project-manager\scripts\addons\Project GLUE\refactoring\move_datablocks.py", line 37, in make_and_remap_target
      mybpy.save_as_mainfile(target_path)
    File "C:\Users\Brad\Documents\blender\project-manager\scripts\addons\Project GLUE\mybpy\__init__.py", line 69, in save_as_mainfile
      ops.wm.save_as_mainfile(filepath=file_path.path, check_existing=False, relative_remap=True)
    File "C:\Program Files\Blender Foundation\Blender 3.6\3.6\scripts\modules\bpy\ops.py", line 113, in __call__
      ret = _op_call(self.idname_py(), None, kw)
  RuntimeError: Error: Cannot open file C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Functions\Integer Math.blend \!\.blend@ for writing: No such file or directory
  
  
  During handling of the above exception, another exception occurred:
  
  Traceback (most recent call last):
    File "C:\Users\Brad\Documents\blender\project-manager\scripts\addons\Project GLUE\ui\execute_action_chain_operator.py", line 36, in execute
      error_message=error_handling.create_bug_report_essentials(e)
  AttributeError: module 'Project GLUE.error_handling' has no attribute 'create_bug_report_essentials'
  Location: C:\Program Files\Blender Foundation\Blender 3.6\3.6\scripts\modules\bpy\ops.py:113
  Exception in module register(): C:\Users\Brad\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\blender_visual_scripting_addon\__init__.py
  Info: Saved "Pop-in Text.blend"
  Error: Cannot open file C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Functions\Integer Math.blend \!\.blend@ for writing: No such file or directory
  Error: Python: Traceback (most recent call last):
    File "C:\Users\Brad\Documents\blender\project-manager\scripts\addons\Project GLUE\ui\execute_action_chain_operator.py", line 26, in execute
      action()
    File "C:\Users\Brad\Documents\blender\project-manager\scripts\addons\Project GLUE\refactoring\move_datablocks.py", line 37, in make_and_remap_target
      mybpy.save_as_mainfile(target_path)
    File "C:\Users\Brad\Documents\blender\project-manager\scripts\addons\Project GLUE\mybpy\__init__.py", line 69, in save_as_mainfile
      ops.wm.save_as_mainfile(filepath=file_path.path, check_existing=False, relative_remap=True)
    File "C:\Program Files\Blender Foundation\Blender 3.6\3.6\scripts\modules\bpy\ops.py", line 113, in __call__
      ret = _op_call(self.idname_py(), None, kw)
  RuntimeError: Error: Cannot open file C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Functions\Integer Math.blend \!\.blend@ for writing: No such file or directory
  
  
  During handling of the above exception, ... truncated
  Blender quit

Environment:
  Blender Version: (3, 6, 1)
  Project Glue Version: (1, 0, 1)
  Operating System: Windows-10-10.0.19045-SP0
  
  Installed Addons:
    io_anim_bvh
    io_curve_svg
    io_mesh_ply
    io_mesh_stl
    io_mesh_uv_layout
    io_scene_fbx
    io_scene_gltf2
    io_scene_obj
    io_scene_x3d
    cycles
    pose_library
    node_wrangler
    space_view3d_copy_attributes
    io_import_images_as_planes
    AI-Render
    development_icon_get
    Shortcutable Reload Scripts
    Project GLUE
    NodePreview
    blender_visual_scripting_addon
  
  Blend Files in Glue Project:
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\1x1x1 Ball - Standalone.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\1x1x1 Ball.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\1x1x1 Cube.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\3x3x8 Selector.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Add Cube Unit Material Attributes.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Arrow.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Audio Wave.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Backdrop.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Box Meter.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Capture Tensor Attributes.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Checkmark.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Connect With Arrow Path.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Conv2D Layer.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Create Filter Window.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Create Text.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Cube Tensor Unit Glass.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Cube Tensor Unit Metal.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Cube Tensor Unit Rubber.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Easing.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Expanding Arrow.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Exploding Tensor.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Eye Pair.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Fade-in Grid.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Filters.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Frame 2D.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Functions - Shader.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Functions.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Geo Nodes PB HSL.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Get Calc Activation.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Ground.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Instanced Tensor.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Large Tensor.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Layer text backup.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Layer.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Lerp.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Mirror.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\PB HSL.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Pop-in Grid.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Pop-in Rotation.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Pop-in Text.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Pop-in.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\primitives.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Range Line.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Set Cube Unit Materials.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Shader - Get Tensor Color.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Shell Text.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Shell.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\speaker.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\string.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Surround Many.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Surround.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Tensor Footprint.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Tensor Placeholder.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Tensor World.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Tensor.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Text Range Line.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Textured Tensor.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Theme.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\To Sphere.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Transformer Key.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Transformer Sinusoid Encoding.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Wireframe Cube.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Word Indexer.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Functions\Clone Object.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Functions\Integer Math.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Functions\Join Objects.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\scenes\Convolution - Theory - Filter Count.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\scenes\Convolution - Theory - Groups.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\scenes\Convolution - Theory - Introduction.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\scenes\Convolution - Theory - Kernel Size.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\scenes\Convolution - Theory - Padding.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\scenes\Convolution - Theory - Stride.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\scenes\demo.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\scenes\GIFs.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\scenes\images-as-tensors.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\scenes\layers.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\scenes\Logo.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\scenes\NN - Introduction.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\scenes\Outro.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\scenes\Scene Template.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\scenes\render\Flat Rainbow.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\scenes\transformers\High Level Task.blend
    C:\Users\Brad\Documents\blender\animated-ai\v2\scenes\transformers\Transformer Block.blend
```

**Comment by animatedai** [comment]
Okay, so maybe there were just multiple threads when the crash happened. There are actually 3 tracebacks. The relevant one (which actually appears last in the list) is:

```
  Traceback (most recent call last):
    File "C:\Users\Brad\Documents\blender\project-manager\scripts\addons\Project GLUE\ui\execute_action_chain_operator.py", line 26, in execute
      action()
    File "C:\Users\Brad\Documents\blender\project-manager\scripts\addons\Project GLUE\refactoring\move_datablocks.py", line 37, in make_and_remap_target
      mybpy.save_as_mainfile(target_path)
    File "C:\Users\Brad\Documents\blender\project-manager\scripts\addons\Project GLUE\mybpy\__init__.py", line 69, in save_as_mainfile
      ops.wm.save_as_mainfile(filepath=file_path.path, check_existing=False, relative_remap=True)
    File "C:\Program Files\Blender Foundation\Blender 3.6\3.6\scripts\modules\bpy\ops.py", line 113, in __call__
      ret = _op_call(self.idname_py(), None, kw)
  RuntimeError: Error: Cannot open file C:\Users\Brad\Documents\blender\animated-ai\v2\primitives\Functions\Integer Math.blend \!\.blend@ for writing: No such file or directory
```

So this is a bug in Project Glue when the user selects a file that Project Glue believe will cause a circular reference. The "/!\\" that  is appended to the file name in the dropdown list is also included in the file name to move to.

**Comment by animatedai** [comment]
Fixed in v1.02
